### PR TITLE
[training_utils] fix: use response_lens.max() instead of offsets().max() for nested tensor max_response_len

### PR DESCRIPTION
**Pull request by dubin555**
### What does this PR do?

Removes the duplicate `_slice_response_from_unpad_output()` function from `losses.py` and replaces its single call site in `value_loss()` with the existing `no_padding_2_padding()` from `padding.py` (already imported at line 26).

The removed function had a bug: it used `response_ids.offsets().max()` for nested tensors, which returns the cumulative offset sum (total token count) instead of the max individual response length. For example, with responses of length `[3, 5, 4]`:
- `offsets() = [0, 3, 8, 12]`
- `offsets().max() = 12` (wrong — total token count)
- `response_lens.max() = 5` (correct — longest response)

Rather than fixing this one line, per maintainer suggestion, we remove the duplicate entirely and use `no_padding_2_padding()` which already handles this correctly.

Search query: https://github.com/verl-project/verl/pulls?q=is%3Apr+_slice_response_from_unpad_output

### Checklist Before Starting

- [x] Search for similar PRs: https://github.com/verl-project/verl/pulls?q=is%3Apr+_slice_response_from_unpad_output
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

The change replaces a buggy private function with the correct public function that is already used throughout the codebase (e.g., in `ray_trainer.py` for `_compute_values`, `_compute_ref_log_prob`, `_compute_log_prob`, and `_compute_entropy`).

`no_padding_2_padding` is covered by existing tests and is exercised in every PPO training run via the model engine path.

### Design & Code Changes

1. **Removed** `_slice_response_from_unpad_output()` (38 lines) — buggy duplicate
2. **Replaced** call in `value_loss()` with `no_padding_2_padding()` — already imported
3. **Removed** unused `import torch.nn.functional as F`

Net change: **-40 lines, +1 line**

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting).

### CODE REVIEW

File: tests/workers/utils/test_slice_response_nested_tensor.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    sequence_lens = prompt_lens + response_lens
+    sequence_offsets = sequence_lens.cumsum(dim=0)
+    assert sequence_offsets[-1].item() == values.shape[0]
+
+    response_list = []
+    for resp_len, seq_offset in zip(response_lens, sequence_offsets, strict=True):
+        pad_size = max_response_len - resp_len
+        response_list.append(F.pad(values[seq_offset - resp_len - 1 : seq_offset - 1], (0, pad_size)))
+
+    return torch.stack(response_list, dim=0)
```
[the remark this answers is not in the document]
Comment (dubin555) [reply tests/workers/utils/test_slice_response_nested_tensor.py]: Good catch! Removed the duplicated `_slice_response_from_unpad_output_fixed` and now importing the actual function from `verl.workers.utils.losses`. Also added a dummy `attention_mask` to the test data since the production function accesses it (even though it's unused in the nested tensor path). Thanks for the suggestion!

**Comment by wuxibin89** [comment]
Nice catch, `_slice_response_from_unpad_output` should be removed and use `no_padding_2_padding` instead, could you help to replace it?

**Comment by dubin555** [comment]
Updated — removed `_slice_response_from_unpad_output` entirely and replaced with `no_padding_2_padding` as suggested. Net change is now -40/+1 lines.
